### PR TITLE
feature/proj-68

### PR DESCRIPTION
**Pull request by Emilo067**
# Pull Request: SentEmail Component

## Description

This pull request introduces the `SentEmail` component, which displays a confirmation message in a modal when an email link is sent to the user. The component is designed to be responsive, ensuring proper display on both desktop and mobile devices.

### Props
- **`email?`**: (optional) The email address to which the confirmation link is sent. If not provided, it defaults to `epam@epam.com`.

### Structure
- **`Modal`**: The parent component that controls the visibility of the modal.
- **`ModalHeader`**: Contains the title and close button of the modal.
- **`ModalContent`**: Wraps the main content of the modal.
- **`Typography`**: Used for text display, including the title and email confirmation message.
- **`ModalFooter`**: Contains the `OK` button that closes the modal.
- **`Button`**: The action button used for closing the modal.
- **`CloseOutline`**: An icon component used to close the modal.

### General Styles
- **Container**: Displays content in a column layout with a gap of `18px` and padding of `0 24px 36px`.
- **Header**: Flexbox layout to space out the title and close icon, with a bottom margin of `30px`.
- **Content**: The modal's content area, with a maximum width of `380px`.
- **Footer**: Aligns the button to the right.
- **Icon**: Sizes the close icon and changes its color on hover and active states.

### Responsive Design
Media queries have been added to ensure proper display on smaller screens:
- **`@media (width <= 768px)`**: 
  - Adjusts the `content` width to `330px` and height to `240px`.
- **`@media (width <= 360px)`**:
  - Adjusts `container` padding to `12px 30px`.
  - Centers the `description` text with a width of `306px`.
  - Adds padding to the `header`.
  - Expands the `button` to full width with a height of `48px`.

### Desktop View
Desktop View

### Mobile View
Mobile View

## Testing
- Tested on both desktop and mobile screen sizes.
- Verified the modal displays the correct email and functions as expected.

---

Please review the component and provide any feedback.

### CODE REVIEW

**Review by EvgeniyaYakunina — APPROVED** [review]

File: src/features/auth/ui/sentEmail/sentEmail.module.scss [hunk 1]
```diff
@@ -0,0 +1,64 @@
+.container {
+  display: flex;
+  flex-direction: column;
+  gap: 18px;
+  padding: 0 24px 36px;
+
+  @media (width <= 360px) {
```
Comment (Lelik7777) [inline src/features/auth/ui/sentEmail/sentEmail.module.scss]: create mixin

create gloabal variable mobile-width

File: src/features/auth/ui/sentEmail/sentEmail.module.scss [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    padding: 0 12px 30px;
+  }
+}
+
+.header {
+  display: flex;
+  justify-content: space-between;
+  margin-bottom: 30px;
+
+  @media (width <= 360px) {
```
Comment (Lelik7777) [inline src/features/auth/ui/sentEmail/sentEmail.module.scss]: use mixin

File: src/features/auth/ui/sentEmail/sentEmail.module.scss [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    padding: 12px;
+  }
+}
+
+.content {
+  width: 100%;
+  max-width: 380px;
+
+  @media (width <= 768px) {
+    width: 100%;
```
Comment (Lelik7777) [inline src/features/auth/ui/sentEmail/sentEmail.module.scss]: ты переопределяешь только max-width

File: src/features/auth/ui/sentEmail/sentEmail.module.scss [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    max-width: 330px;
+    height: 240px;
```
Comment (Lelik7777) [inline src/features/auth/ui/sentEmail/sentEmail.module.scss]: высоту никогда не харкодим

File: src/features/auth/ui/sentEmail/sentEmail.module.scss [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+  }
+}
+
+.footer {
+  display: flex;
+  justify-content: end;
+
+  button {
+    @media (width <= 360px) {
```
Comment (Lelik7777) [inline src/features/auth/ui/sentEmail/sentEmail.module.scss]: use mixin

File: src/features/auth/ui/sentEmail/sentEmail.module.scss [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+      width: 100%;
+      height: 48px;
```
Comment (Lelik7777) [inline src/features/auth/ui/sentEmail/sentEmail.module.scss]: высоту не харкодим

File: src/features/auth/ui/sentEmail/sentEmail.module.scss [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+    }
+  }
+}
+
+.description {
+  @media (width <= 360px) {
```
Comment (Lelik7777) [inline src/features/auth/ui/sentEmail/sentEmail.module.scss]: use mixin

File: src/features/auth/ui/sentEmail/sentEmail.module.scss [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+    width: 306px;
```
Comment (Lelik7777) [inline src/features/auth/ui/sentEmail/sentEmail.module.scss]: все же лучше width and max-width

File: src/features/auth/ui/sentEmail/sentEmail.tsx [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+          <ModalClose>
+            <CloseOutline className={classNames.icon} />
+          </ModalClose>
+        </ModalHeader>
+        <div className={classNames.container}>
+          <Typography className={classNames.description} variant={'text16'}>
+            We have sent a link to confirm your email to {finalEmail}
+          </Typography>
+          <ModalFooter className={classNames.footer}>
+            <Button onClick={onClickHandler}>OK</Button>
```
Comment (Lelik7777) [inline src/features/auth/ui/sentEmail/sentEmail.tsx]: вопрос обсувждаем, но, как по мне, нажатие на эту кнопку просто дублирует эффект нажатия на крестик и просто закрывает модалку

Comment (Lelik7777) [inline src/features/auth/ui/sentEmail/sentEmail.tsx]: вопрос обсувждаем, но, как по мне, нажатие на эту кнопку просто дублирует эффект нажатия на крестик и просто закрывает модалку

File: src/features/auth/ui/sentEmail/sentEmail.module.scss (hunk 1 above)
Comment (Emilo067) [reply src/features/auth/ui/sentEmail/sentEmail.module.scss]: исправил

File: src/features/auth/ui/sentEmail/sentEmail.module.scss (hunk 2 above)
Comment (Emilo067) [reply src/features/auth/ui/sentEmail/sentEmail.module.scss]: исправил

File: src/features/auth/ui/sentEmail/sentEmail.module.scss (hunk 3 above)
Comment (Emilo067) [reply src/features/auth/ui/sentEmail/sentEmail.module.scss]: исправил

File: src/features/auth/ui/sentEmail/sentEmail.module.scss (hunk 4 above)
Comment (Emilo067) [reply src/features/auth/ui/sentEmail/sentEmail.module.scss]: исправил

File: src/features/auth/ui/sentEmail/sentEmail.module.scss (hunk 5 above)
Comment (Emilo067) [reply src/features/auth/ui/sentEmail/sentEmail.module.scss]: исправил

File: src/features/auth/ui/sentEmail/sentEmail.module.scss (hunk 8 above)
Comment (Emilo067) [reply src/features/auth/ui/sentEmail/sentEmail.module.scss]: исправил

**Pull request merged**